subsys/bluetooth/services/ots_l2cap.c

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/printk.h>
+#include <sys/byteorder.h>
+#include <zephyr.h>
+#include <init.h>
+
+#include <net/buf.h>
+
+#include "ots_l2cap_internal.h"
+
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(bt_gatt_ots, CONFIG_BT_GATT_OTS_LOG_LEVEL);
+
+/* According to BLE specification Assigned Numbers that are used in the
+ * Logical Link Control for protocol/service multiplexers.
+ */
+#define BLE_GATT_OTS_L2CAP_PSM	0x0025
+
+/* Maximum size of TX buffer and its payload. */
+#define MAX_TX_BUF_SIZE		256
```

Comment:
Could this perhaps be derived from the CONFIG_BT_GATT_OTS_L2CAP_CHAN_RX_MTU or L2CAP MPS?

Reply:
This is TX direction and RX_MTU can be different. I don't see any L2CAP MPS configuration.

I would rather not make it a configurable option since the OTS user doesn't need to know how data is actually sent over L2CAP. If he sends packets that is longer than 256 bytes (via ots_obj_read), then such a packet will be fragmented by this layer.